jetty-client/src/main/java/org/eclipse/jetty/client/util/ByteBufferContentProvider.java

```diff
@@ -76,9 +76,8 @@ public ByteBuffer next()
                 try
                 {
                     ByteBuffer buffer = buffers[index];
-                    buffers[index] = buffer.slice();
```

Comment:
So the original code does take a slice, but replaces the original buffer with the slice.   Why?
If there is no reason, the update looks good.